client/sections-middleware.js

```diff
@@ -114,3 +114,11 @@ export const setupRoutes = () => {
 		}
 	}
 };
+
+if ( module.hot ) {
+	module.hot.accept( './sections', () => {
+		const updatedSections = require( './sections' ).default;
+		receiveSections( updatedSections );
+		setupRoutes();
+	} );
```

Comment:
I'm wondering in what situations does the `sections` hot update happen. When merely changing a React component, only the one module with that component should hot-update. But it seems that for you, it sometimes leads to a `sections` update, too? Did you investigate why did that happen? It would be nice to compare the incoming `sections.hot-update.js` module with the existing `sections.js`. What has changed?

Reply:
I'm not sure why it's happening. If I do these steps:
* `yarn && yarn start`
* Visit `http://calypso.localhost:3000/home/<mysite>`
* Make this change:
```diff
diff --git a/client/my-sites/customer-home/main.jsx b/client/my-sites/customer-home/main.jsx
index e88a776681..ce47a98c29 100644
--- a/client/my-sites/customer-home/main.jsx
+++ b/client/my-sites/customer-home/main.jsx
@@ -110,7 +110,7 @@ const Home = ( {
                        <FormattedHeader
                                brandFont
                                headerText={ translate( 'My Home' ) }
-                               subHeaderText={ translate( 'Your hub for posting, editing, and growing your site.' ) }
+                               subHeaderText={ translate( 'Your hub2 for posting, editing, and growing your site.' ) }
                                align="left"
                                hasScreenOptions
                        />
```

In my console I see:
```
[webpack] hot updated 2 modules: webpack-client.ts:94:13
./my-sites/customer-home/main.jsx webpack-client.ts:97:15
./sections.js
```

In my terminal I see:
```
webpack building...
<i> [webpack-dev-middleware] wait until bundle finished: /home/testsite5mmr.word press.com
<i> [webpack-dev-middleware] wait until bundle finished: /nostats.js?_ut=wpcom%3Auser_id&_ui=73763986
<i> [webpack-dev-middleware] wait until bundle finished: /version?1684511999671
webpack built cec808f1f61c2467932f in 10135ms
assets by status 168 MiB [cached] 2105 assets
assets by path . 2.29 MiB
  assets by info 288 KiB [immutable]
    asset boot_common_js.194cf1bb8654a5a5279b.hot-update.js 265 KiB [emitted] [immutable] [hmr]
    asset home.194cf1bb8654a5a5279b.hot-update.js 20.4 KiB [emitted] [immutable] [hmr] (name: home)
    asset runtime.194cf1bb8654a5a5279b.hot-update.js 2.94 KiB [emitted] [immutable] [hmr] (name: runtime)
    asset runtime.194cf1bb8654a5a5279b.hot-update.json 55 bytes [emitted] [immutable] [hmr]
  asset home.js 1.03 MiB [emitted] (name: home)
  asset boot_common_js.js 787 KiB [emitted]
  asset vendors-node_modules_socket_io-client_lib_index_js.js 146 KiB [emitted] (id hint: vendors)
  asset runtime.js 74.7 KiB [emitted] (name: runtime)
cached modules 52.2 MiB (javascript) 16.8 MiB (asset) 12.7 MiB (css/mini-extract) [cached] 15761 modules
runtime modules 55.4 KiB 24 modules
javascript modules 41.3 KiB
  cacheable modules 41 KiB
    modules by path ./ 28.6 KiB
      ./sections.js 18.9 KiB [code generated]
      ./my-sites/customer-home/main.jsx 9.74 KiB [built] [code generated]
    modules by path ../node_modules/ 12.4 KiB
      ../node_modules/socket.io-client/lib/manager.js 12.2 KiB [code generated]
      ../node_modules/engine.io-client/lib/index.js 140 bytes [code generated]
  ../packages/components/src/animation/features/ lazy ^.*$ namespace object 160 bytes [built]
  ./signup/steps/ lazy ^\.\/.*$ include: signup\/steps\/[0-9a-z/-]+\/index\.[j%7Ct]sx$ exclude: signup\/steps\/[0-9a-z/-]+\/test\/index\.[j%7Ct]sx$ chunkName: async-load-signup-steps-[request] namespace object 160 bytes [built]
webpack 5.68.0 compiled successfully in 10135 ms

Ready! All assets are re-compiled. Have fun!

```

And my new code block runs.
I'm not sure how to see what's different; I tried this form of logging:
```diff
diff --git a/client/sections-middleware.js b/client/sections-middleware.js
index f4b8a2f698..558bf1c9c6 100644
--- a/client/sections-middleware.js
+++ b/client/sections-middleware.js
@@ -11,6 +11,7 @@ import isSectionEnabled from './sections-filter';
 import { receiveSections, load } from './sections-helper';
 import { pathToRegExp } from './utils';
 
+console.log(sections);
 receiveSections( sections );
 
 function activateSection( section, context ) {
@@ -118,6 +119,7 @@ export const setupRoutes = () => {
 if ( module.hot ) {
        module.hot.accept( './sections', () => {
                const updatedSections = require( './sections' ).default;
+               console.log(updatedSections);
                receiveSections( updatedSections );
                setupRoutes();
        } );
```

Then using chrome devtools to assign each to a global variable, copy() it, paste them to files and diff them, and they were the exact same.

If I look at the network tab, the important requests seem to be to `http://calypso.localhost:3000/calypso/evergreen/home.5f3b16f21430644f17b8.hot-update.js` and `http://calypso.localhost:3000/calypso/evergreen/boot_common_js.5f3b16f21430644f17b8.hot-update.js`. I suppose the offender here is it sending me a new boot_common_js. But it's a huge string that's `evaled()` that I can't make much sense of - about 275K of text.